tests/lib/rules/camelcase.js

```diff
@@ -337,6 +337,18 @@ ruleTester.run("camelcase", rule, {
             code: "__option_foo__ = 0;",
             options: [{ allow: ["__option_foo__"] }]
         },
+        {
+            code: "__option_foo__ = 0; user_id = 0; foo = 1",
+            options: [{ allow: ["__option_foo__", "_id$"] }]
+        },
+        {
+            code: "var$key = 0;",
+            options: [{ allow: ["__option_foo__", "\\$key"] }]
+        },
```

Comment:
This code would be valid without options, maybe you intended to use some other variable name?